lib/read.js

```diff
@@ -74,7 +80,7 @@ function read (req, res, next, parse, debug, options) {
 
   // read body
   debug('read body')
-  getBody(stream, opts, function (error, body) {
+  getBody(stream, opts, preserveAsyncContext('GetBody', function (error, body) {
```

Comment:
Should not the `getBody` function itself be preserving the async context? Wouldn't adding it within `getBody` function directly preserve the context for every dependency on npm that uses it instead of needing to work through every one of them (like this module) to add this? It seems quite burdensome for all the dependencies of that module if this is is true. Because by that logic, why don't you do this around the call to `bodyParser` itself?

Reply:
Yep, adding it directly to getBody is probably better.